_perlin.c

```diff
@@ -173,12 +173,12 @@ noise3(float x, float y, float z, const int repeatx, const int repeaty, const in
 	int ii = (int)fmodf(i + 1,  repeatx);
 	int jj = (int)fmodf(j + 1, repeaty);
 	int kk = (int)fmodf(k + 1, repeatz);
-	i = (i & 255) + base;
-	j = (j & 255) + base;
-	k = (k & 255) + base;
-	ii = (ii & 255) + base;
-	jj = (jj & 255) + base;
-	kk = (kk & 255) + base;
+	i = ((i + base) & 255);
+	j = ((j + base) & 255);
+	k = ((k + base) & 255);
+	ii = ((ii + base) & 255);
+	jj = ((jj + base) & 255);
+	kk = ((kk + base) & 255);
 
```

Comment:
From what I understand you're capping the result to 255. This can be misleading to the user since it's a hidden operation : the user could expect different results with larger values for base and not get them.

I'd recommend throwing an error or at least a warning.